Task_1_3_1

ViktorPetrovichBarinov opened the pull request.

ViktorPetrovichBarinov committed:
calculateJumpTable

ViktorPetrovichBarinov committed:
test

ViktorPetrovichBarinov committed:
123

ViktorPetrovichBarinov committed:
11

ViktorPetrovichBarinov committed:
123

ViktorPetrovichBarinov committed:
111

BIGNIKi commented on Task_1_3_1/src/main/java/org/example/KMP.java:
```diff
@@ -0,0 +1,123 @@
+
+        try {
+            File fileDirectory = new File(fileName);
+            BufferedReader br = new BufferedReader(
```
Try with resources используй

ViktorPetrovichBarinov replied:
+

BIGNIKi commented on Task_1_3_1/src/main/java/org/example/KMP.java:
```diff
@@ -0,0 +1,123 @@
+        try {
+            File fileDirectory = new File(fileName);
+            BufferedReader br = new BufferedReader(
+                    new InputStreamReader(
```
Давай еще сделаешь чтение файла из ресурсов. Полезно будет разобраться, что такое ресурсы в джаве, где ждет их gradle и что gradle с ними делает при сборке.

ViktorPetrovichBarinov replied:
+

BIGNIKi commented on Task_1_3_1/src/test/java/org/example/KMPTest.java:
```diff
@@ -0,0 +1,107 @@
+/**
+ * Some text.
+ */
+public class KMPTest {
```
Сделай еще тестовый метод, который генерит большой файл (15-20 гигов), ищет в нем подстроку (какую-нибудь небольшую, которая находится в самом конце файла). Также программно удалить этот файл. (как junit тест его не делай, кто знает как на это github workflow отреагирует)

ViktorPetrovichBarinov replied:
+

BIGNIKi reviewed (changes requested):
Защитил

ViktorPetrovichBarinov committed:
Поправляю замечания

Добавил использование try-with-resources. Переделал, чтобы файлы читались из ресурсов.

ViktorPetrovichBarinov committed:
add bigDataTest

ViktorPetrovichBarinov committed:
checkstyle test

ViktorPetrovichBarinov committed:
checkstyle 2
